rust/storage/src/config.rs

```diff
@@ -17,6 +19,28 @@ pub enum StorageConfig {
     AdmissionControlledS3(AdmissionControlledS3StorageConfig),
 }
 
+#[derive(Deserialize, Debug, Clone)]
+pub enum ObjectStoreType {
```

Comment:
Why can we not reuse `S3CredentialsConfig`

Reply:
The name threw me off.  Can I rename it `CredentialsConfig` for long term viability?